tests/sub_port_interfaces/conftest.py

```diff
@@ -275,9 +275,12 @@ def apply_route_config_for_port(request, duthost, ptfhost, define_sub_ports_conf
     port_type = define_sub_ports_configuration['port_type']
     subnet = define_sub_ports_configuration['subnet']
 
-    # Get additional port for configuration of SVI port
+    # Get additional port for configuration of SVI port or L3 RIF
     if 'svi' in request.param:
-        dut_ports, ptf_ports = get_port(duthost, ptfhost, 1, port_type, dut_ports.values())
+        interface_num = 1
+    else:
+        interface_num = 2
+    dut_ports, ptf_ports = get_port(duthost, ptfhost, interface_num, port_type, dut_ports.values(), exclude_sub_interface_ports=True)
```

Comment:
Are we assuming a 1-1 mapping for dut_ports and ptf_ports in this code flow?  question is based on L291 where we do a zip(). Look at the return statement in get_port(). there clearly cannot be a 1-1 mapping since both are dictionaries. Btw this question is not based off the sub port changes in this PR, just making sure the generic flow is correct.

Reply:
Yes, the reason to get new ports for the L3 test is that the test case now have an issue configuring a RIF over a port that has sub interfaces, the RIF won't work. `dut_ports` was a dictionary with two items(https://github.com/Azure/sonic-mgmt/blob/1c85de1bbe302748d822c7ca22f0b4fcf39db004/tests/sub_port_interfaces/conftest.py#L109), we are just trying to find two ports that has no sub interfaces to replace the original `dut_ports`